Automattic/wp-calypso
# BackupScheduleSetting: add next scheduled backup time on Latest Backup section

Related to https://github.com/Automattic/jetpack-backup-team/issues/700

## Proposed Changes

* Add next scheduled backup time Latest Backup section and include a `Modify` action that will point to the `Daily backup time schedule` setting. This message will be under the `jetpack/backup-schedule-setting` feature flag.
  * Move `convertHourToRange` function to a utils file
  * Add `useNextBackupSchedule` hook that will be responsible for getting the date and time range.
  * Add `NextScheduledBackup` component that will render the new message in the `Latest backup` section.
* Replace `Restore to this point` button with a [SplitButton](https://wpcalypso.wordpress.com/devdocs/design/split-button) including all the items from the actions toolbar (Download backup, View files, etc.). 
  * Decouple actions (Download, Restore, etc.) into separate components in `activity-card/toolbar/actions` directory
  * Enable `RestoreButton` component to render as a `SplitButton` when `useSplitButton` prop is passed.

| Before | After |
|---|---|
| ![CleanShot 2024-10-23 at 00 21 23@2x](https://github.com/user-attachments/assets/32f8d1c6-6f81-4643-b9a8-c32da2f0259b) | ![CleanShot 2024-10-23 at 00 22 59@2x](https://github.com/user-attachments/assets/adbe24c5-9914-4f3c-8937-26e41bc43408) |
| ![CleanShot 2024-10-23 at 00 24 34@2x](https://github.com/user-attachments/assets/a980ac74-6b56-4350-b0b5-44e0820fb012) | ![CleanShot 2024-10-23 at 00 25 08@2x](https://github.com/user-attachments/assets/7f4d5604-d5c9-427b-a5b4-d1cc461b8101) |

> [!NOTE]
> The changes on the `Restore to this point` will be live once this PR is merged. Only the `Next full backup` message will be under a feature flag. This is how it will looks when the feature flag is not enabled:
> 
> ![CleanShot 2024-10-23 at 00 39 46@2x](https://github.com/user-attachments/assets/20b5ae27-311b-4838-a458-be0cefbde479)

## Why are these changes being made?


https://github.com/Automattic/jetpack-backup-team/issues/700

## Testing Instructions



* Spin up a Jetpack Cloud live branch.
* Pick a site with Jetpack VaultPress Backup plan.
* Navigate to the `Backup` page.
* If you have a backup from Today, you should see when the day and time range of the next scheduled backup, and the `Restore to this point` split button should look similar to the screenshots shared above.
* Try clicking on `Modify` and ensure it is redirecting to the Settings page.
* Try updating your schedule time and go back to the `Backup` page to ensure the change is reflected there.
* Ensure all buttons works as expected.

## Pre-merge Checklist
